src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioRazorParser.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Text;
+using ITextBuffer = Microsoft.VisualStudio.Text.ITextBuffer;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public class VisualStudioRazorParser : IDisposable
```

Comment:
The goal of making it abstract would just be to prevent it being constructable. Since this PR doesn't include the lifetime stuff, it's good enough for now as-is 👍